index.js

```diff
@@ -52,4 +52,129 @@ const state = {
+
+  // add the text to the button
+  storeBtn.innerText = 'Add to cart'
+
+  // add event listener to button which checks if
+  // the item is not in the cart ! (bang). If not then
+  // it will add the item while giving it a quantity
+  // of one. If it is already in the cart then
+  // the item quantity will increase by 1
+  storeBtn.addEventListener('click', () => {
+    if (!state.cart.includes(state.items[i])) {
+      state.items[i].quantity = 1
+      state.cart.push(state.items[i])
+    } else {
+    state.items[i].quantity++
+  }
+  generateCart()
+  })
+
+  // append the icon, button and list item to the unordered list
+  storeLi.append(storeImg, storeBtn)
+  storeUl.append(storeLi)
+}
+
+function generateCart() {
+  // clear cart html
+  cartUl.innerHTML = ''
+
+  // for loop to generate cart
+  for (let i = 0; i < state.cart.length; i++) {
+
+    // create elements
+    const cartLi = document.createElement('li')
+    const cartItemImg = document.createElement('img')
+    const cartItemName = document.createElement('p')
+    const removeBtn = document.createElement('button')
+    const cartQuantity = document.createElement('span')
+    const addBtn = document.createElement('button')
+
+    // add a source to the img using string interpolation
+    // add a class to the img to apply correct styling
+    cartItemImg.setAttribute('src', `assets/icons/${state.cart[i].id}.svg`)
+    cartItemImg.setAttribute('class', '.cart--item-icon')
+
+    // add text to paragraph(name of item)
+    cartItemName.innerText = `${state.cart[i].name}`
+
+    // remove button add classes and text
+    removeBtn.setAttribute('class', 'center quantity-btn remove-btn')
+    removeBtn.innerText = '-'
+  
+    // add button add classes and text
+    addBtn.setAttribute('class', 'center quantity-btn add-btn')
+    addBtn.innerText = '+'
+
+
+    // show quantity for each cart item
+    cartQuantity.innerText = `${state.cart[i].quantity}`
+    
+    // add an event listener to the minus button
+    // to take off 1 per click. if the value hits
+    // 0 then the item will be removed
+    removeBtn.addEventListener('click', () => {
+      if (state.cart[i].quantity > 1) {
+        state.cart[i].quantity--
+      } else {
+        state.cart.splice(i, 1)
+      }
+      generateCart()
+    })
+
+    // add an event listener to increase the quantity
+    // every time the button has been clicked
+    addBtn.addEventListener('click', () => {
+      state.cart[i].quantity++
+      generateCart()
+    })
+
+    // append all the elements to the list item then append
+    // the list item to the unordered list
+    cartLi.append(cartItemImg, cartItemName, removeBtn, cartQuantity, addBtn)
+    cartUl.append(cartLi)
+  }
+
+  // generate the total cost for the cart
+  // using the reduce method
+  // !for some reason having this nested in the for loop
+  // worked apart from the fact that if the cart became
+  // empty it wouldn't reset to 0.
```

Comment:
I'm not familiar with the accumulator but have you tried another method so that it would reset to 0 when the cart is empty? (Assuming you still have this issue)

Reply:
I only had this issue when it was nested within the for loop, taking it outwith the scope of that worked, Not sure why it almost worked but it's fine now